tools/testsys/src/vmware_k8s.rs

```diff
@@ -0,0 +1,242 @@
+            .crd_input
+            .existing_crds(&labels, &["testsys/type", "testsys/controlPlaneEndpoint"])
+            .await?;
+
+        let cluster_version =
+            K8sVersion::from_str(cluster_input.crd_input.variant.version().context(
+                error::MissingSnafu {
+                    item: "K8s version".to_string(),
+                    what: "aws-k8s variant".to_string(),
+                },
+            )?)
+            .map_err(|_| error::Error::K8sVersion {
+                version: cluster_input.crd_input.variant.to_string(),
+            })?;
+
+        let vsphere_k8s_crd = VSphereK8sClusterConfig::builder()
+            .name(cluster_input.cluster_name)
+            .set_labels(Some(labels))
+            .control_plane_endpoint_ip(control_plane_endpoint)
+            .creation_policy(CreationPolicy::IfNotExists)
+            .version(cluster_version)
+            .ova_name(self.image_id(cluster_input.crd_input)?)
+            .tuf_repo(
+                cluster_input
+                    .crd_input
+                    .tuf_metadata()
+                    .context(error::InvalidSnafu {
+                        what: "TUF repo information is required for vmware cluster creation.",
+                    })?,
+            )
+            .vcenter_host_url(&self.datacenter.vsphere_url)
+            .vcenter_datacenter(&self.datacenter.datacenter)
+            .vcenter_datastore(&self.datacenter.datastore)
+            .vcenter_network(&self.datacenter.network)
+            .vcenter_resource_pool(&self.datacenter.resource_pool)
+            .vcenter_workload_folder(&self.datacenter.folder)
+            .mgmt_cluster_kubeconfig_base64(&self.encoded_mgmt_cluster_kubeconfig)
+            .set_conflicts_with(Some(existing_clusters))
+            .destruction_policy(DestructionPolicy::OnTestSuccess)
+            .image(
+                cluster_input
+                    .crd_input
+                    .images
+                    .vsphere_k8s_cluster_resource_agent_image
+                    .as_ref()
+                    .expect(
+                        "The default VSphere K8s cluster resource provider image URI is missing.",
+                    ),
+            )
+            .set_image_pull_secret(
+                cluster_input
+                    .crd_input
+                    .images
+                    .testsys_agent_pull_secret
+                    .to_owned(),
+            )
+            .set_secrets(Some(cluster_input.crd_input.config.secrets.clone()))
+            .privileged(true)
+            .build(cluster_input.cluster_name)
+            .map_err(|e| error::Error::Build {
+                what: "VSphere k8s cluster CRD".to_string(),
+                error: e.to_string(),
+            })?;
+        Ok(CreateCrdOutput::NewCrd(Box::new(Crd::Resource(
+            vsphere_k8s_crd,
+        ))))
+    }
+
+    /// Create a vSphere VM provider CRD to launch Bottlerocket VMs on the cluster created by
```

Comment:
```suggestion
    /// Create a VSphere VM provider CRD to launch Bottlerocket VMs on the cluster created by
```

And same for all occurrences, there are a few.

Reply:
Please don't capitalize the v.

Reply:
I did because this PR has vSphere/VSphere, I didn't know which one was correct! :sweat:

Reply:
As a former VMware employee, I can say you can usually go with the one that you wouldn't expect. :D